public/pages/Indices/components/AddPolicyModal/AddPolicyModal.tsx

```diff
@@ -37,39 +42,51 @@ interface AddPolicyModalProps {
 
 interface AddPolicyModalState {
   isLoading: boolean;
-  selectedPolicies: PolicyOption[];
+  selectedPolicy: PolicyOption | null;
+  hasRolloverAction: boolean;
   policyOptions: PolicyOption[];
+  rolloverAlias: string;
 }
 
 export default class AddPolicyModal extends Component<AddPolicyModalProps, AddPolicyModalState> {
   state: AddPolicyModalState = {
     isLoading: false,
-    selectedPolicies: [],
+    selectedPolicy: null,
+    hasRolloverAction: false,
     policyOptions: [],
+    rolloverAlias: "",
   };
 
-  async componentDidMount() {
+  async componentDidMount(): Promise<void> {
     await this.onPolicySearchChange("");
   }
 
   onAddPolicy = async (): Promise<void> => {
     try {
-      const { selectedPolicies } = this.state;
+      const { selectedPolicy, hasRolloverAction, rolloverAlias } = this.state;
       const {
         onClose,
         indices,
         services: { indexService },
       } = this.props;
-      if (selectedPolicies.length !== 1) {
-        toastNotifications.addDanger(`There are no selected indices`);
+      if (!indices) {
+        toastNotifications.addDanger("There are no selected indices");
```

Comment:
I thought we use toast for API response or any other kind of errors, why are we using toastNotifactions for validation ? why can't we leverage form's validation ?